src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs

```diff
@@ -1,38 +1,41 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
 using System;
 using System.Collections.Generic;
-using System.IO;
+using System.Diagnostics;
 using System.Linq;
-using NuGet.ProjectModel;
-using NuGet.Packaging.Core;
-using Microsoft.Build.Framework;
-using Microsoft.Build.Utilities;
 
 namespace Microsoft.NET.Build.Tasks
 {
     internal class ProjectContext
     {
         private readonly LockFile _lockFile;
         private readonly LockFileTarget _lockFileTarget;
-        private readonly string _platformLibraryName;
         internal HashSet<PackageIdentity> PackagesToBeFiltered { get; set; }
 
-        public bool IsPortable { get; }
+        public bool IsFrameworkDependent { get; }
         public LockFileTargetLibrary PlatformLibrary { get; }
 
         public LockFile LockFile => _lockFile;
         public LockFileTarget LockFileTarget => _lockFileTarget;
 
-        public ProjectContext(LockFile lockFile, LockFileTarget lockFileTarget, string platformLibraryName)
+        public ProjectContext(LockFile lockFile, LockFileTarget lockFileTarget, LockFileTargetLibrary platformLibrary, bool isFrameworkDependent)
         {
+            Debug.Assert(lockFile != null);
+            Debug.Assert(lockFileTarget != null);
+            if (isFrameworkDependent)
+            {
+                Debug.Assert(platformLibrary != null);
+            }
```

Comment:
I kind of like the readability of the way it is. It is very straight-forward to understand (at least to me).  The compiler will remove the `if` statement in release mode, since it is empty, so it isn't like it will affect the outputted assembly code.

Do you have a strong opinion on this?